src/tests/test_demographics.py

```diff
@@ -59,3 +61,27 @@ def test_stepwise_age_distribution(stepwise_age_module):
     for i in range(0, N):
         age_count = sum((ages >= age_boundaries[i]) & (ages < age_boundaries[i+1]))
         assert pytest.approx(age_count, rel=0.1) == age_dist[i]*count
+
+
+def test_date_permanent(default_module):
+    """Check that that death is not recorded again in future steps."""
+    population_data = pd.DataFrame({
+        'date_of_death': [None, datetime.today(), datetime.today() - timedelta(days=1)],
+        'age': [30, 20, 50],
+        'sex': [SexType.Female, SexType.Female, SexType.Male]
+    })
+    new_deaths = default_module.determine_deaths(population_data)
+    # Find who already had a date of death, and check that they are not marked
+    # as having died in this time step.
+    assert not new_deaths[population_data.date_of_death.notnull()].any()
```

Comment:
Surely this always passes because no one is dead when the population is created, and the population isn't evolved before calculating `new_deaths`, so `population_data.date_of_death.notnull()` would always be empty?

Reply:
This isn't using a `Population`, but rather the specific data frame defined here, which has two people marked as dead already.